src/Microsoft.Health.Operations/Serialization/OperationIdJsonConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Operations.Serialization;
+
+/// <summary>
+/// Represents a strict <see cref="JsonConverter{T}"/> for operation IDs.
+/// </summary>
+/// <remarks>
+/// Unlike the typical <see cref="Guid"/> deserialization, the <see cref="OperationIdJsonConverter"/>
+/// only accepts strings that can be parsed using the <see cref="OperationId.FormatSpecifier"/>.
+/// </remarks>
+public sealed class OperationIdJsonConverter : JsonConverter<Guid>
```

Comment:
As a note, there are two serialization libraries in play: json.net and system.text.json.

For all types internal to our activities and orchestrations, we rely on json.net (aka newtonsoft) as that is the library used within the Azure Functions framework. However, our code (like the serialization between the client and server for our web apps), will use the newer System.Text.Json.